ui/src/app/applications/components/utils.tsx

```diff
@@ -127,6 +132,129 @@ export const ComparisonStatusIcon = ({status, resource, label}: {status: appMode
     );
 };
 
+export function showDeploy(resource: string, appContext: AppContext) {
+    appContext.apis.navigation.goto('.', {deploy: resource});
+}
+
+export function renderResourceMenu(
+    resource: ResourceTreeNode,
+    application: appModels.Application,
+    appContext: AppContext,
+    appChanged: BehaviorSubject<appModels.Application>,
+    getApplicationActionMenu: () => any
+): React.ReactNode {
+    let menuItems: Observable<ActionMenuItem[]>;
+    if (isAppNode(resource) && resource.name === application.metadata.name) {
+        menuItems = Observable.from([getApplicationActionMenu()]);
+    } else {
+        const isRoot = resource.root && nodeKey(resource.root) === nodeKey(resource);
+        const items: MenuItem[] = [
```

Comment:
Looks like changes introduced in https://github.com/argoproj/argo-cd/commit/e1990faf678c1fad47df3dcbaa990f297842f5f2#diff-d3ef7de562f9a4a34a9a0b05a112955fdecdd0102c3faae5eeb03a195091a5e4 got lost. UI should show prompts only for "managed" resources.